polyfill/lib/calendar.mjs

```diff
@@ -333,53 +332,1354 @@ function resolveNonLunisolarMonth(calendarDate) {
+    Av: { leap: 12, regular: 11, monthCode: 'M11', days: 30 },
+    Elul: { leap: 13, regular: 12, monthCode: 'M12', days: 29 }
+  },
+  getMonthCode(year, month) {
+    if (this.inLeapYear({ year })) {
+      return month === 6 ? 'M5L' : `M${month < 6 ? month : month - 1}`;
+    } else {
+      return `M${month}`;
+    }
+  },
+  adjustCalendarDate(calendarDate, cache, overflow = 'constrain', fromLegacyDate = false) {
+    // The current Intl implementation skips index 6 (Adar I) in non-leap years
+    let { year, eraYear, month, monthCode, day, monthExtra } = calendarDate;
+    if (year === undefined) year = eraYear;
+    if (eraYear === undefined) eraYear = year;
+    if (fromLegacyDate) {
+      // DateTimeFormat.formatToParts returns either a numeric string (e.g. "4")
+      // or the name of a month (e.g. "Tevet"), even when the month format is
+      // set to 'numeric. Even running the same exact code in the Chrome dev
+      // tools debugger yields different results!  I assume it's a caching bug
+      // somewhere in the Intl code. I've not been able to reproduce the cause
+      // of the divergent behavior. But given that this implementation is a
+      // prototype and will be thrown away in a few months, for now we'll defend
+      // against this behavior here by handling either a numeric string or a
+      // name.
+      if (monthExtra) {
+        const monthInfo = this.months[monthExtra];
+        if (!monthInfo) throw new RangeError(`Unrecognized month from formatToParts: ${monthExtra}`);
+        month = this.inLeapYear({ year }) ? monthInfo.leap : monthInfo.regular;
+      }
+      monthCode = this.getMonthCode(year, month);
+      const result = { year, month, day, era: undefined, eraYear, monthCode };
+      return result;
+    } else {
+      // When called without input coming from legacy Date output, simply ensure
+      // that all fields are present.
+      this.validateCalendarDate(calendarDate);
+      if (month === undefined) {
+        if (monthCode.endsWith('L')) {
+          if (monthCode !== 'M5L') throw new RangeError(`Hebrew leap month must have monthCode M5L, not ${monthCode}`);
+          month = 6;
+          if (!this.inLeapYear({ year })) {
+            if (overflow === 'reject') {
+              throw new RangeError(`Hebrew monthCode M5L is invalid in year ${year} which is not a leap year`);
+            } else {
+              // constrain to last day of previous month (Av)
+              month = 5;
+              day = 30;
+              monthCode = 'M5';
+            }
+          }
+        } else {
+          month = monthCodeNumberPart(monthCode);
+          // if leap month is before this one, the month index is one more than the month code
+          if (this.inLeapYear({ year }) && month > 6) month++;
+        }
+      }
+      if (monthCode === undefined) monthCode = this.getMonthCode(year, month);
+      return { ...calendarDate, day, month, monthCode, year, eraYear };
```

Comment:
It shouldn't cause any harm. For internal-only processing, I generally followed the rule of always filling in `eraYear === year` for non-era-using calendars in order to avoid special-case code in cases where `eraYear` is used.  As you noted, I I don't think this is observable behavior.

Reply:
> Also, should the hebrew calendar have constant era 'am' or 'ah'?

I assumed that `hasEra` should be `true` for calendars whose default `.toLocaleDateString('en-US-u-ca-...')` output contains an era, and `false` for calendars where it doesn't. The latter group is only ISO, Hebrew, and Chinese/Dangi. So that's why there's no era in Hebrew. But this is also really easy to change. @sffc, @Manishearth - do you guys have any opinions about which calendars should and should not use eras?